frontend-react/src/utils/TestUtils.ts

```diff
@@ -20,4 +20,35 @@ const mockEvent = (mock?: Partial<any>) => {
     };
 };
 
-export { mockEvent };
+function conditionallySupressConsole(...matchers: string[]) {
```

Comment:
Thanks for adding this!  Super useful!

Reply:
That said, I think we should use caution when using this.  It looks like most/all of the warnings/errors in this pull request shouldn't be escalated to the test log, but I'm wary of overusing this just to create a silenced log.  Our first inclination should be to look into the why warnings/errors are getting printed out for a given test and silence them only if there's nothing actually fishy with said test.  We can also explicitly `expect` them if they're intentional, of course.

Not asking for any changes -- just something for the team to keep in mind!

Reply:
Very true. This is only to be used for scenarios where we are expecting console messages (error handling testing and such) to reduce noise to see real console messages that need addressing. I was debating on whether the return should be a tuple or object containing the spy objects so that it's possible to check the call counts and such.

Reply:
No worries!  I think it's fine to keep this a separate function named as such so it gets attention called to it during code review.  Otherwise, we can manually set up spies when we're intending to catch warnings/errors.

One small, small thing, though -- I think `suppress` has two `p`s 😆